src/track/macroaction.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "engine/engine.h"
+#include "proto/macro.pb.h"
+namespace proto = mixxx::track::io;
+
+/// A MacroAction is the building block of a Macro.
+/// It contains a position as well as the action to be taken at that position.
+///
+/// Note that currently only jumps to a target position are available, but that
+/// is subject to change.
+class MacroAction {
+  public:
+    enum class Type : uint32_t {
+        Jump = 0
+    };
+
+    MacroAction(double sourceFramePos, double targetFramePos)
```

Comment:
I'm questioning if we should really save sample/frame positions in the db. I don't want to run into the same problems we have with Cues/Beatmap/Beatgrid. I think Ideally, we'd have a lightweight MacroInfo class similar to CueInfo that stores positions in milliseconds, not samples/frames. It could have a method like:

```cpp
Macro macro = macroInfo.getMacroWithStreamInfo(streamInfo);
```

This would definitely make it easier to handle import/export from/to Serato Metadata without having to postpone the import until the stream info is available.

@xerus2000 @uklotzde What do you think?

Reply:
oof, does make sense, but I wanted to merge this week and not introduce major architectural reworks ^^
also, these things should be brought up in Zulip where there's more room for such bigger discussions and they won't get lost, rather than a temporary Github review comment

Anyways, why do you want to name it `MacroInfo` and not stay with `MacoAction`?

Reply:
That wasn't related to this class, I was talking about the macro class. Just to distinguish the time-based class from the frame-based one.